frontend/src/BudgetModeDisplay.tsx

```diff
@@ -20,23 +25,40 @@ export function BudgetModeDisplay({
     currentFleetSupply,
     maxFleetSupply,
     unitCounts,
+    isFighterUpgraded,
+    shipCapacityUsed,
+    maxShipCapacity,
+    spaceDockFighterBonus,
 }: Props) {
+    const shipCapacityRemaining =
+        maxShipCapacity +
+        spaceDockFighterBonus -
+        shipCapacityUsed -
+        getFighterCount(unitCounts);
+
     return [
         {
             label: "Resources Remaining",
             value: resourceBudget - getTotalCost(unitCounts),
         },
         {
-            label: "Capacity Remaining",
+            label: "Production Capacity Remaining",
             value: capacityBudget - getTotalCapacity(unitCounts),
         },
         {
             label: "Fleet Supply Remaining",
             value: getFleetSupplyRemaining(
                 currentFleetSupply,
                 maxFleetSupply,
-                unitCounts
+                unitCounts,
+                isFighterUpgraded ? Math.max(0 - shipCapacityRemaining, 0) : 0
             ),
         },
+        {
+            label: "Ship Capacity Remaining",
+            value: isFighterUpgraded
```

Comment:
As above, check if you prefer:
`isFighterUpgraded && shipCapacityRemaining < 0 ? 0 : shipCapacityRemaining`

Reply:
Prefer yours, updated! I was playing around with a `zeroFloor(num: number) { return Math.max(num, 0); }` util to abstract out the `Math.max(x, 0)`s everywhere but it still ended up being hard to digest, I give up 🥲

Reply:
Oooh and, did one more thing, the latest commit reduces the confusion even more - [47d71a6](https://github.com/sirrus233/twilight-imperium-helper/pull/3/commits/47d71a69edfcbdd57984328a200222209e56887f)